django_x509/base/models.py

```
@@ -349,48 +360,92 @@ def _import(self):
+            bc = ca_cert.extensions.get_extension_for_class(x509.BasicConstraints).value
+            if not bc.ca:
+                raise ValidationError(
+                    _(
+                        "The selected CA is not authorized to sign certificates"
+                        "(BasicConstraints)."
+                    )
+                )
+        except x509.ExtensionNotFound:
+            raise ValidationError(
+                _("The selected CA is missing BasicConstraints and cannot sign.")
+            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)